lib/snippets/deploy-to-gke

```diff
@@ -53,16 +57,19 @@ class GKEDeployment
 
   private
 
-  def enforce_required_parmas
-    [@namespace, @gcloud_project, @gcloud_deployment, @current_sha].each do |required_param|
+  def enforce_required_params
+    [@namespace, @gcloud_project, @gcloud_deployment, @current_sha, @key_file].each do |required_param|
       raise ArgumentError, "#{required_param} is required" unless required_param && !required_param.empty?
     end
   end
 
+  def preliminary_check
+    raise FatalDeploymentError, "Project config missing at #{@key_file}" unless File.file?(@key_file)
+    raise FatalDeploymentError, "#{@template_path} doesn't exist" unless File.directory?(@template_path)
+    raise FatalDeploymentError, "#{@template_path} doesn't have files with postfix .yml or .yml.erb" unless Dir.entries('/tmp').select {|file| file =~ /\.yml(.erb)?$/}.size > 0
+  end
+
   def apply_all_templates
-    unless File.directory?(@template_path)
-      raise FatalDeploymentError, "#{@template_path} doesn't exist"
-    end
     found = 0
```

Comment:
I felt like it would be nice if we ever had problems matching weird names or whatnot. Eg. if you have n files there and suddenly only see n-x being applied. I dont feel strongly, but just added info really